src/python/pants/backend/native/subsystems/native_build_step_settings_base.py

```diff
@@ -9,23 +9,55 @@
 from pants.subsystem.subsystem import Subsystem
 
 
-class NativeBuildStepSettingsBase(Subsystem, MirroredTargetOptionMixin):
+class NativeBuildStepSettingsBase(Subsystem, NativeBuildSettings, MirroredTargetOptionMixin):
 
   mirrored_option_to_kwarg_map = {
     'fatal_warnings': 'fatal_warnings',
+    'compiler_option_sets': 'compiler_option_sets',
   }
 
   @classmethod
   def register_options(cls, register):
     super(NativeBuildStepSettingsBase, cls).register_options(register)
 
-    # TODO: implement compiler_option_sets as an interface to platform/host-specific optimization
-    # flags!
-    register('--fatal-warnings', type=bool, default=True, fingerprint=True, advanced=True,
-             help='The default for the "fatal_warnings" argument for targets of this language.')
+    register('--fatal-warnings-enabled-args', advanced=True, type=list, fingerprint=True,
```

Comment:
The changes to the fatal-warnings option should not be necessary here: can leave it as is, and then it can be deprecated in a followup.